js/src/ui/Form/AddressSelect/addressSelect.js

```diff
@@ -15,262 +15,799 @@
+import keycode, { codes } from 'keycode';
+
+import CloseIcon from 'material-ui/svg-icons/navigation/close';
+import TextFieldUnderline from 'material-ui/TextField/TextFieldUnderline';
+
+import ParityBackground from '~/ui/ParityBackground';
+import IdentityIcon from '~/ui/IdentityIcon';
+import InputAddress from '~/ui/Form/InputAddress';
+import Tags from '~/ui/Tags';
 import { fromWei } from '~/api/util/wei';
 
 import styles from './addressSelect.css';
 
-export default class AddressSelect extends Component {
+class AddressSelect extends Component {
   static contextTypes = {
-    api: PropTypes.object.isRequired
-  }
+    muiTheme: PropTypes.object.isRequired
+  };
 
   static propTypes = {
+    // Required props
     onChange: PropTypes.func.isRequired,
 
+    // Redux props
+    accountsInfo: PropTypes.object,
     accounts: PropTypes.object,
-    allowInput: PropTypes.bool,
     balances: PropTypes.object,
     contacts: PropTypes.object,
     contracts: PropTypes.object,
+    tokens: PropTypes.object,
+    wallets: PropTypes.object,
+
+    // Optional props
+    allowInput: PropTypes.bool,
     disabled: PropTypes.bool,
     error: PropTypes.string,
     hint: PropTypes.string,
     label: PropTypes.string,
-    tokens: PropTypes.object,
-    value: PropTypes.string,
-    wallets: PropTypes.object
-  }
+    value: PropTypes.string
+  };
 
-  state = {
-    autocompleteEntries: [],
-    entries: {},
-    addresses: [],
+  static defaultProps = {
     value: ''
-  }
+  };
 
-  // Cache autocomplete items
-  items = {}
+  nameRefs = {};
 
-  entriesFromProps (props = this.props) {
-    const { accounts = {}, contacts = {}, contracts = {}, wallets = {} } = props;
+  state = {
+    copied: null,
+    expanded: false,
+    focused: false,
+    focusedCat: null,
+    focusedItem: null,
+    inputFocused: false,
+    inputValue: '',
+    left: 0,
+    top: 0,
+    values: []
+  };
 
-    const autocompleteEntries = [].concat(
-      Object.values(wallets),
-      'divider',
-      Object.values(accounts),
-      'divider',
-      Object.values(contacts),
-      'divider',
-      Object.values(contracts)
-    );
+  componentWillMount () {
+    this.setValues();
 
-    const entries = {
-      ...wallets,
-      ...accounts,
-      ...contacts,
-      ...contracts
-    };
+    window.addEventListener('resize', this.handleTagsOpacity);
+  }
 
-    return { autocompleteEntries, entries };
+  componentDidMount () {
+    this.setPosition();
   }
 
-  shouldComponentUpdate (nextProps, nextState) {
-    const keys = [ 'error', 'value' ];
+  componentWillReceiveProps (nextProps) {
+    if (this.values && this.values.length > 0) {
+      return;
+    }
 
-    const prevValues = pick(this.props, keys);
-    const nextValues = pick(nextProps, keys);
+    this.setValues(nextProps);
+  }
 
-    return !isEqual(prevValues, nextValues);
+  componentWillUnmount () {
+    window.removeEventListener('resize', this.handleTagsOpacity);
```

Comment:
I didn't really get it. This is needed because this component hooks up on the resize event for displaying the tags correctly depending on the screen size. When the component unmount, the hook is thus no more needed. I think we should start using `componentWillUnmount` more than we do, especially when modifying the state within the Component (eg. using async actions that would modify the state => if the Component is unmounted, it would throw an error if we don't handle it correctly).